src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -355,6 +351,23 @@ public AgentSettings LoadSettings()
             var serviceControlManager = HostContext.GetService<ILinuxServiceControlManager>();
             serviceControlManager.GenerateScripts(settings);
 #endif
+
+            // Get and apply Tags in case agent is configured against Machine Group
+            bool needToAddTags = agentProvider.GetAddTagsRequired(command);
```

Comment:
as of now we are not able to handle the case where user input is empty string and default value is also an empty string